Package.swift

```diff
@@ -4,120 +4,170 @@
 import PackageDescription
 import CompilerPluginSupport
 
-// Availability Macros
-let availabilityMacros: [SwiftSetting] = [
-    "FoundationPreview 0.1:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4",
-    "FoundationPredicate 0.1:macOS 14, iOS 17, tvOS 17, watchOS 10",
-    "FoundationPreview 0.2:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4",
-    "FoundationPreview 0.3:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4",
-    "FoundationPredicate 0.3:macOS 14, iOS 17, tvOS 17, watchOS 10",
-    "FoundationPreview 0.4:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4",
-    "FoundationPredicate 0.4:macOS 14, iOS 17, tvOS 17, watchOS 10",
-].map { .enableExperimentalFeature("AvailabilityMacro=\($0)") }
+// To develop this package on Apple platforms, set this to true
+let developmentOnDarwin = false
 
-let package = Package(
-    name: "FoundationPreview",
-    platforms: [.macOS("13.3"), .iOS("16.4"), .tvOS("16.4"), .watchOS("9.4")],
-    products: [
-        // Products define the executables and libraries a package produces, and make them visible to other packages.
-        .library(name: "FoundationPreview", targets: ["FoundationPreview"]),
-        .library(name: "FoundationEssentials", targets: ["FoundationEssentials"]),
-        .library(name: "FoundationInternationalization", targets: ["FoundationInternationalization"]),
-    ],
-    dependencies: [
-        .package(
-            url: "https://github.com/apple/swift-collections",
-            revision: "d8003787efafa82f9805594bc51100be29ac6903"), // on release/1.1
-        .package(
-            url: "https://github.com/apple/swift-foundation-icu",
-            exact: "0.0.5"),
-        .package(
-            url: "https://github.com/apple/swift-syntax.git",
-            from: "509.0.2")
-    ],
-    targets: [
-        // Foundation (umbrella)
-        .target(
-            name: "FoundationPreview",
-            dependencies: [
+#if canImport(Darwin)
+let useSystemFoundation = !developmentOnDarwin
+#else
+// On non Darwin platforms always use package
+let useSystemFoundation = false
+#endif
```

Comment:
I don't recommend doing it this way since this will break cross compilation from Darwin to another platform. `swift-crypto` just defines a variable `isDevelopment = false` here and package maintainers manually set that `true`. Setting it to `true` allows to run the code in the package on Darwin platforms. In general using `#if canImport` in `Package.swift` is not a good idea cc @MaxDesiatov

Reply:
+1 to what Franz said, this check will just break all packages depending on `swift-foundation` when cross-compiling.

Reply:
Keep in mind that for some platforms one can only cross-compile, e.g. Wasm/WASI, embedded etc. Presence of `#if canImport` or `#if os` checks in `Package.swift` make this package and everything that depends on it unusable for such platforms.